### PR TITLE
Add a redirect history

### PR DESCRIPTION
**Pull request by stefanruijsenaars**
Includes coding standard fixes in https://github.com/guzzle/guzzle/pull/1183

### CODE REVIEW

File: tests/RedirectMiddlewareTest.php [hunk 1]
```diff
@@ -130,6 +130,30 @@ public function testAddsRefererHeader()
...
+        $request = new Request('GET', 'http://example.com?a=b');
+        $promise = $handler($request, [
+            'allow_redirects' => [
+                  'track_redirects' => true,
+            ],
+        ]);
+        /** @var ResponseInterface $response */
+        $response = $promise->wait(TRUE);
+
+        $this->assertEquals(['http://example.com?a=b', 'http://example.com', 'http://example.com/foo'], $response->getHeader('X-Guzzle-Redirect'));
```
Comment (GrahamCampbell) [inline tests/RedirectMiddlewareTest.php]: this line is quite long

**Comment by mtdowling** [comment]
You should really branch off the existing work that this originated from so that the original author's commits are still in the git history.

File: src/RedirectMiddleware.php [hunk 2]
```diff
@@ -98,7 +99,18 @@ public function checkRedirect(
...
         }
 
-        return $this($nextRequest, $options);
+        /** @var PromiseInterface|ResponseInterface $response */
+        $options['_previous_uris'][] = $request->getUri();
+        $response = $this($nextRequest, $options);
+
+        // Add headers to be able to track history of redirects.
+        if (!empty($options['allow_redirects']['track_redirects'])) {
+            $response = $response->wait(TRUE);
```
Comment (mtdowling) [inline src/RedirectMiddleware.php]: We should not block here. It should be a promise.

**Comment by mtdowling** [comment]
#1183 was merged with modifications so this is no longer necessary.

**Pull request closed without merging**